src/lib/djerba/plugins/fusion/plugin.py

```diff
@@ -67,22 +80,155 @@ def sort_by_actionable_level(row):
             }
             gene_info = []
             treatment_opts = []
+
+        # Processing fusions and generating blob URLs
+        tsv_file_path = wrapper.get_my_string(fc.ARRIBA_PATH)
+        base_dir = (directory_finder(self.log_level, self.log_path).get_base_dir())
+        fusion_dir = os.path.join(base_dir, "plugins", "fusion")
+        json_template_path = os.path.join(fusion_dir, fc.JSON_TO_BE_COMPRESSED)
+        output_dir = self.workspace.get_work_dir()
+        unique_fusions = list({item["fusion"] for item in results[fc.BODY]})
+        fusion_url_pairs = []
+
+        failed_fusions = 0
+
+        for fusion in unique_fusions:
+            try:
+                fusion, blurb_url = self.process_fusion(config, fusion, tsv_file_path, json_template_path, output_dir)
+                fusion_url_pairs.append([fusion, blurb_url])
+
+            except FusionProcessingError as e:
+                self.logger.error(f"Skipping fusion {fusion}: {e}")
+                failed_fusions += 1
+
+        if failed_fusions > 0:
+            self.logger.warning(f"{failed_fusions} fusions failed out of {len(unique_fusions)}.")
+
+        # Save the fusion-URL pairs to a CSV file
+        output_tsv_path = os.path.join(output_dir, 'fusion_blurb_urls.tsv')
+        with open(output_tsv_path, 'w', newline='') as tsvfile:
+            writer = csv.writer(tsvfile, delimiter='\t')
+            writer.writerow(['Fusion', 'Whizbam URL'])
+            writer.writerows(fusion_url_pairs)
+
         data = self.get_starting_plugin_data(wrapper, self.PLUGIN_VERSION)
         data[core_constants.RESULTS] = results
         data[core_constants.MERGE_INPUTS]['gene_information_merger'] = gene_info
         data[core_constants.MERGE_INPUTS]['treatment_options_merger'] = treatment_opts
         return data
 
+    def process_fusion(self, config, fusion, tsv_file_path, json_template_path, output_dir):
+        wrapper = self.get_config_wrapper(config)
+
+        # Validate and parse the fusion format
+        match = re.match(r"(.+)::(.+)", fusion)
+        if not match:
+            raise FusionProcessingError(f"No valid fusion found for {fusion}. Ensure the format is gene1::gene2.")
```

Comment:
Log the error message with `self.logger.error` before doing `raise FusionProcessingError`